modules/oe_subscriptions_digest/tests/src/Functional/UserDigestTestBase.php

```diff
@@ -0,0 +1,179 @@
+
+    $select = $assert_session->selectExists('Notifications frequency');
+
+    $this->assertEquals([
+      'Send immediately' => 'Send immediately',
+      'message_digest:daily' => 'Daily',
+      'message_digest:weekly' => 'Weekly',
+    ], $this->getOptions($select));
+
+    MessageDigestInterval::create([
+      'id' => 'bi_weekly',
+      'label' => 'Bi-weekly',
+      'interval' => '2 weeks',
+    ])->save();
+    $this->drupalGet($path);
+    $this->assertEquals([
+      'Send immediately' => 'Send immediately',
+      'message_digest:daily' => 'Daily',
+      'message_digest:weekly' => 'Weekly',
+      'message_digest:bi_weekly' => 'Bi-weekly',
+    ], $this->getOptions($select));
+
+    MessageDigestInterval::create([
+      'id' => 'monthly',
+      'label' => 'Monthly',
+      'interval' => '1 month',
+    ])->save();
+    $this->drupalGet($path);
+    $this->assertEquals([
+      'Send immediately' => 'Send immediately',
+      'message_digest:daily' => 'Daily',
+      'message_digest:weekly' => 'Weekly',
+      'message_digest:bi_weekly' => 'Bi-weekly',
+      'message_digest:monthly' => 'Monthly',
+    ], $this->getOptions($select));
+
+    $select->selectOption('Daily');
+    $assert_session->buttonExists('Save')->press();
+    $assert_session->statusMessageContains('Your preferences have been saved', 'status');
+
+    $user_storage = \Drupal::entityTypeManager()->getStorage('user');
+    $user_storage->resetCache();
+    /** @var \Drupal\user\UserInterface $user */
+    $user = $user_storage->load($user->id());
+    $this->assertEquals('message_digest:daily', $user->get('message_digest')->value);
+  }
+
+  /**
+   * Tests the flagging digest updating.
+   *
+   * @param \Drupal\user\UserInterface $user
+   *   The user for the test.
+   */
+  protected function doTestFlaggingDigest(UserInterface $user): void {
+    $this->drupalCreateContentType([
+      'type' => 'page',
+      'name' => 'Page',
+    ]);
+    $pages_flag = $this->createFlagFromArray([
+      'id' => 'subscribe_page',
+      'entity_type' => 'node',
+      'bundles' => ['page'],
+    ]);
+    $another_flag = $this->createFlagFromArray([
+      'id' => 'another_flag',
+      'entity_type' => 'node',
+      'bundles' => [],
+    ]);
+    $page_one = $this->drupalCreateNode([
+      'type' => 'page',
+      'status' => 1,
+    ]);
+    $page_two = $this->drupalCreateNode([
+      'type' => 'page',
+      'status' => 1,
+    ]);
+    FieldConfig::create([
+      'field_name' => 'message_digest',
+      'entity_type' => 'flagging',
+      'label' => 'Frequency',
+      'bundle' => 'subscribe_page',
+      'description' => '',
+      'required' => FALSE,
+      'settings' => [],
+    ])->save();
+    FieldConfig::create([
+      'field_name' => 'message_digest',
+      'entity_type' => 'flagging',
+      'label' => 'ABC',
+      'bundle' => 'another_flag',
+      'description' => '',
+      'required' => FALSE,
+      'settings' => [],
+    ])->save();
```

Comment:
Not needed.

Reply:
Removed. For default flags provided by message_digest_ui we have the frequency field, but the duplicated with the email prefix we don't so I added a field creation on those in the same hook.

Reply:
Missunderstood on my side, changing